File: E_ThreeTierReact/frontend/src/service/ProductService.js
import axios from 'axios'

const API_URL = import.meta.env.VITE_API_URL;

class ProductService{

    saveProduct(product){
        return axios.post(API_URL+"/", product);
    }

    getAllProducts(){
        return axios.get(API_URL+"/")
    }

    deleteProduct(id){
        return axios.delete(API_URL+`/delete/${id}`);
    }

    updateProduct(product, id){
        axios.post(`${API_URL}/product/${id}`, product);
    }
}

export default new ProductService();

File: E_ThreeTierReact/E_ThreeTierReact/src/main/java/com/om/service/ProductService.java
package com.om.service;

import java.util.List;

import com.om.Entity.Product;

public interface ProductService {

	public Product saveProduct(Product product);
	
	public Product editProduct(Product product, Integer id);
	
	public List<Product> getAllProducts();
	
	public Product getProductById(Integer id);
	
	public String deleteProduct(Integer id);
	
	
}
